accepted/2021/winforms/streamline-application-bootstrap.md

```diff
@@ -0,0 +1,302 @@
+
+
+## Scenarios and User Experience
+
+<!--
+Provide examples of how a user would use your feature. Pick typical scenarios
+first and more advanced scenarios later.
+
+Ensure to include the "happy path" which covers what you expect will satisfy the
+vast majority of your customer's needs. Then, go into more details and allow
+covering more advanced scenarios. Well designed features will have a progressive
+curve, meaning the effort is proportional to how advanced the scenario is. By
+listing easy things first and more advanced scenarios later, you allow your
+readers to follow this curve. That makes it easier to judge whether your feature
+has the right balance.
+
+Make sure your scenarios are written in such a way that they cover sensible end-
+to-end scenarios for the customer. Often, your feature will only cover one
+aspect of an end-to-end scenario, but your description should lead up to your
+feature and (if it's not the end result) mention what the next steps are. This
+allows readers to understand the larger picture and how your feature fits in.
+
+If you design APIs or command line tools, ensure to include some sample code on
+how your feature will be invoked. If you design UI, ensure to include some
+mock-ups. Do not strive for completeness here -- the goal of this section isn't
+to provide a specification but to give readers an impression of your feature and
+the look & feel of it. Less is more.
+-->
+
+Refer to [Design](#Design) section below.
+
+![projectconfiguration](https://user-images.githubusercontent.com/4403806/119747720-aa0a1b80-bed6-11eb-8247-45b1668d4867.gif)
+
+## Requirements
+
+### Goals
+
+<!--
+Provide a bullet point list of aspects that your feature has to satisfy. This
+includes functional and non-functional requirements. The goal is to define what
+your feature has to deliver to be considered correct.
+
+You should avoid splitting this into various product stages (like MVP, crawl,
+walk, run) because that usually indicates that your proposal tries to cover too
+much detail. Keep it high-level, but try to paint a picture of what done looks
+like. The design section can establish an execution order.
+-->
+
+1. The new bootstrap experience (this includes source generators and analyzers) must come inbox with the Windows Desktop SDK, and be available without any additional references or NuGet packages from get go (i.e. `dotnet new winforms && dotnet build`).<br/>Related: [dotnet/designs#181](https://github.com/dotnet/designs/pull/181)
+1. The new bootstrap API must only work for Windows applications projects (i.e. `OutputType = WinExe`). These projects have an entry point, where an app is initialised; and they also specify an application manifest, if there is one.
+2. The new bootstrap must be correctly generated for .NET Framework and .NET projects taking in consideration available 
```

Comment:
Does this bootstrap actually need to support .NET Framework at all? I'm wondering why we would change things there?

Reply:
Here I've been thinking about the multi-targeting scenarios. I'm still trying to work out how the analyzers weave into the SDK though, may be this point isn't possible after all.

Reply:
I think I've been overthinking this. If analyzers are placed under dotnet\packs\Microsoft.WindowsDesktop.App.Ref\\X.Y.<version\>\ref\netX.Y then there is no backwards compatibility or multitargeting scenarios to consider - the analyzers will be imported by the Windows Desktop SDK for the given TFM. And if there was no analyzers available for .NET Framework or .NET 5.0 SDKs, then those can't/won't be imported.